app/services/twilio/incoming_message_service.rb

```diff
@@ -106,15 +106,25 @@ def additional_attributes
   end
 
   def attach_files
-    return if params[:MediaUrl0].blank?
+    num_media = params[:NumMedia].to_i
+    return if num_media.zero?
 
-    attachment_file = download_attachment_file
+    num_media.times do |i|
+      attach_single_file(i)
+    end
+  end
+
+  def attach_single_file(index)
+    media_url = params[:"MediaUrl#{index}"]
+    content_type = params[:"MediaContentType#{index}"]
+    return if media_url.blank?
```

Comment:
One general suggestion overall is that instead of passing the `index` we can pass the `MediaUrl#1` directly. In this particular function we are using the `index` to get the content type, but that is not necessary either (explained in the next comment)

This should make the functions a bit cleaner, having to build the param name in every function can be error prone